### DISCUSSION TITLE
Calling javascript funtion on pressing Native Windows Xaml Button

### DISCUSSION THREAD

**Discussion by hunainhumail-purevpn** [microsoft/react-native-windows#11448] [Q&A] [answered]
I have a button which is created on XAML in UWP Windows app. I want to achieve functionality that when I press on Xaml button then the javascript function from React Native will be called and axios api will be called from the function.
The response should then get back to the Native View or console.

Note: I have achieved this functionality in Android, IOS and MacOS. I want to achieve it in windows. The problem I am facing is that I am getting null in ReactContext. 

In Android, I was able to get the current React Context through this statement:

JAVA:
 ReactContext reactContext = mReactInstanceManager.getCurrentReactContext();
 
 For this I need ReactInstanceManager. But in windows, I am unable to see this ReactInstanceManager. 
 
 I have creating a NativeButtonHandler.cs class in which I want to achieve this funtionality. I am calling it from ButtonOnPress inside Main.xaml.cs
 
 I want to use EventEmitter in windows for this functionality. I see that the ReactContext has a function named EmitJSEvent. But my windows app is crashing on pressing the button as it is getting reactcontext = null. 

Any help would be appreciated.

Thanks

**Comment by acoates-ms** [comment]
You can get the `ReactContext` within a NativeModule/TurboModule using the `REACT_INIT` macro.  Or if you want it on a more global way from the event args on `ReactNativeHost.InstanceSettings.InstanceCreated` or `InstanceLoaded` event.

**Reply by hunainhumail-purevpn** [reply]
Can you please explain the answer more precisely? Let me show you where I am stucked..

Here is my Main.xaml.cs file:

public sealed partial class MainPage : Page
    {
        public MainPage()
        {
            this.InitializeComponent();
            var app = Application.Current as App;
            reactRootView.ReactNativeHost = app.Host;
        }

        private void Button_Click(object sender, RoutedEventArgs e)


        {
            var nativeBridge = new NativeButtonHandler(reactContext);   <----- Here I need to pass the reactContext @acoates-ms 
            nativeBridge.OnXamlButtonClick();


            //var reactInstance = (Application.Current as App).Host.ReactInstance;

        }
    }

**Reply by acoates-ms** [reply, marked as answer]
Something like:

In App.xaml.cs:

void InstanceCreated(object sender, InstanceCreatedEventArgs args)  
{  
  this.ReactContext = args.Context;
}  

public IReactContext ReactContext {get; set;}

In App()

add:
InstanceSettings.InstanceCreated += InstanceCreated

Then in your button handler you can use (Application.Current as App).ReactContext

**Reply by hunainhumail-purevpn** [reply]
@acoates-ms  Thank You so much! Its working like a charm now!
